scripts/Engineering/texture/TextureUtils.py

```diff
@@ -398,13 +444,30 @@ def fit_all_peaks(wss: Sequence[str], peaks: Sequence[float], peak_window: float
             bg_func_name = "LinearBackground"
 
             func_generator = TexturePeakFunctionGenerator([])  # don't fix any parameters
+            parameters_to_tie = () if not parameters_to_tie else parameters_to_tie
             initial_function, md_fit_kwargs, intensity_estimates = func_generator.get_initial_fit_function_and_kwargs_from_specs(
-                ws, peak, (xmin, xmax), ("A", "B"), peak_func_name, bg_func_name
+                ws, peak, (xmin, xmax), parameters_to_tie, peak_func_name, bg_func_name
             )
 
-            Fit(
+            fit_object = Fit(
                 Function=initial_function,
                 CostFunction="Least squares",
+                Output="first_fit",
+                MaxIterations=10,  # if it hasn't fit in 10 it is likely because the texture has the peak missing
```

Comment:
10 iterations seems pretty harsh, even for the initial fit! One possibility is also to screen spectra with `np.sum(y)/np.sqrt(np.sum(e**2)) < i_over_sigma_thresh` - i.e. assuming background is zero, just checking there are some counts in the data!